src/Bootstrap/BootstrapByConfigFile.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * @author Irfan Durmus (http://github.com/irfan) <irfandurmus@gmail.com>
+ *
+ * Moved some initialization logic from src/functions.php to here, since
+ * putting application logic in public functions which callable without
+ * any restriction is not good.
+ *
+ * We do not need any inheritance or interface implementation here,
+ * it's just simple POPO class.
+ */
+
+namespace Deployer\Bootstrap;
+
+use Deployer\Type\DotArray;
+use Deployer\Cluster\ClusterBuilder;
+use Deployer\Server\Builder;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @property string $configFile
+ */
+class BootstrapByConfigFile
+{
+    
+    /**
+     * @var string | null $configFile
+     */
+    public $configFile = null;
+
+    /**
+     * @var string | null $configFileContent
```

Comment:
I think your IDE complaining because the default value. It looks okay in the workflow. the Yaml::parse running in a try/catch block. So, in case there is a null, it will fall to the catch block. 

I can add something like
if ($this->configFile !== null) 
in the try block, but it makes no sense.